homeassistant/components/automation/__init__.py

```diff
@@ -795,6 +795,22 @@ def _log_callback(self, level: int, msg: str, **kwargs: Any) -> None:
         """Log helper callback."""
         self._logger.log(level, "%s %s", msg, self.name, **kwargs)
 
+    async def _async_trigger_if_enabled(
+        self,
+        run_variables: dict[str, Any],
+        context: Context | None = None,
+        skip_condition: bool = False,
+    ) -> ScriptRunResult | None:
+        """Trigger automation if enabled.
+
+        If the trigger starts but has a delay, the automation will be triggered
+        when the delay has passed so we need to make sure its still enabled before
+        executing the action.
+        """
+        if not self._is_enabled:
+            return None
+        return await self.async_trigger(run_variables, context, skip_condition)
```

Comment:
Separate function because we still allow automations to be from the service even if they are disabled

Reply:
https://github.com/home-assistant/core/blob/968de08e4b90bc31059a7fb8a0829c8100cb87fe/homeassistant/components/automation/__init__.py#L307